contrib/initramfs/scripts/zfs.in

```diff
@@ -411,15 +411,21 @@ decrypt_fs()
 
 		# Determine dataset that holds key for root dataset
 		ENCRYPTIONROOT="$(get_fs_value "${fs}" encryptionroot)"
+		KEYLOCATION="$(${ZFS} get -H -o value keylocation "${ENCRYPTIONROOT}")"
 
 		# If root dataset is encrypted...
 		if ! [ "${ENCRYPTIONROOT}" = "-" ]; then
 			KEYSTATUS="$(get_fs_value "${ENCRYPTIONROOT}" keystatus)"
 			# Continue only if the key needs to be loaded
 			[ "$KEYSTATUS" = "unavailable" ] || return 0
 			TRY_COUNT=3
+
+			# If keylocation is not "prompt", do not load key with a prompt helper
```

Comment:
It'd be nice to avoid the somewhat double negative. Could this be: If keylocation is not "prompt", load the key directly.

Reply:
Or perhaps similar to above, "If key is stored in a file, do not prompt.".

Reply:
Yup, I agree. Changed to match